utils/chunk-tests.js

```diff
@@ -0,0 +1,152 @@
+        .replace(rootPath, '')
+        .split(path.sep);
+
+      const packagePath =
+        path.join(packageDir, shortPackageName) + ',' + packageName;
+      testsToRun[packagePath] = testsToRun[packagePath] || {};
+      testsToRun[packagePath][scriptName] =
+        testsToRun[packagePath][scriptName] || [];
+      testsToRun[packagePath][scriptName].push(testPath);
+    });
+
+  const chunkedTests = Object.entries(testsToRun).flatMap(
+    ([packagePathAndName, scriptNames]) => {
+      const [packagePath, packageName] = packagePathAndName.split(',');
+      return Object.entries(scriptNames).flatMap(([scriptName, testPaths]) => {
+        return intoChunks(NUMBER_OF_CHUNKS, testPaths).flatMap(
+          (chunk, chunkNumber, allChunks) => {
+            const runners = runnersMap.get(scriptName) || ['ubuntu-latest'];
+
+            return runners.map(runner => {
+              return {
+                runner,
+                packagePath,
+                packageName,
+                scriptName,
+                testPaths: chunk.map(testFile =>
+                  path.relative(
+                    path.join(__dirname, '../', packagePath),
+                    testFile
+                  )
+                ),
+                chunkNumber: chunkNumber + 1,
+                allChunksLength: allChunks.length,
+              };
+            });
+          }
+        );
+      });
+    }
+  );
+
+  return chunkedTests;
+}
+
+async function turbo(args) {
+  const chunks = [];
+  try {
+    await new Promise((resolve, reject) => {
+      const spawned = child_process.spawn(`yarn`, ['turbo', '--', ...args], {
+        cwd: path.resolve(__dirname, '..'),
+        env: {
+          ...process.env,
+          YARN_SILENT: '1',
+        },
+      });
+      spawned.stdout.on('data', data => {
+        chunks.push(data);
+        process.stderr.write(data);
+      });
+      spawned.stderr.on('data', data => {
+        process.stderr.write(data);
+      });
+      spawned.on('close', code => {
+        if (code !== 0) {
+          reject(new Error(`Turbo exited with code ${code}`));
+        } else {
+          resolve();
+        }
+      });
+    });
+    return Buffer.concat(chunks);
+  } catch (e) {
+    console.error(e.message);
+    process.exit(1);
+  }
+}
+
+/**
+ * @template T
+ * @param {number} totalChunks maximum number of chunks
+ * @param {T[]} values
+ * @returns {T[][]}
+ */
+function intoChunks(totalChunks, arr) {
+  const chunkSize = Math.max(
+    MINIMUM_PER_CHUNK,
+    Math.ceil(arr.length / totalChunks)
+  );
+  const chunks = [];
+  for (let i = 0; i < totalChunks; i++) {
+    chunks.push(arr.slice(i * chunkSize, (i + 1) * chunkSize));
+  }
+  return chunks.filter(x => x.length > 0);
+}
+
+async function main() {
+  try {
+    const chunks = await getChunkedTests(process.argv[2]);
+    // TODO: pack and build the runtimes for each package and cache it so we only deploy it once
```

Comment:
Is this work tracked somewhere?

Reply:
This util is from [the runtimes repo](https://github.com/vercel/runtimes/blob/main/utils/chunk-tests.js) so not sure if we're tracking this somewhere related to that.